packages/react-native-drawer-layout/src/views/Drawer.native.tsx

```diff
@@ -217,53 +222,76 @@ export function Drawer({
-      'worklet';
-      touchX.value = event.x;
-      translationX.value = startX.value + event.translationX;
-      gestureState.value = event.state;
-    })
-    .onEnd((event, success) => {
-      'worklet';
-      gestureState.value = event.state;
-
-      if (!success) {
-        runOnJS(onGestureAbort)();
-      }
-
-      const nextOpen =
-        (Math.abs(event.translationX) > SWIPE_MIN_OFFSET &&
-          Math.abs(event.translationX) > swipeMinVelocity) ||
-        Math.abs(event.translationX) > swipeMinDistance
-          ? drawerPosition === 'left'
-            ? // If swiped to right, open the drawer, otherwise close it
-              (event.velocityX === 0 ? event.translationX : event.velocityX) > 0
-            : // If swiped to left, open the drawer, otherwise close it
-              (event.velocityX === 0 ? event.translationX : event.velocityX) < 0
-          : open;
-
-      toggleDrawer(nextOpen, event.velocityX);
-      runOnJS(onGestureFinish)();
-    })
-    .activeOffsetX([-SWIPE_MIN_OFFSET, SWIPE_MIN_OFFSET])
-    .failOffsetY([-SWIPE_MIN_OFFSET, SWIPE_MIN_OFFSET])
-    .hitSlop(hitSlop)
-    .enabled(drawerType !== 'permanent' && swipeEnabled);
-
-  if (pan && configureGestureHandler) {
-    pan = configureGestureHandler(pan);
-  }
+  const pan = React.useMemo(() => {
+    let panGesture = Gesture?.Pan()
+      .onBegin((event) => {
+        'worklet';
+        startX.value = translationX.value;
+        gestureState.value = event.state;
+        touchStartX.value = event.x;
+      })
+      .onStart(() => {
+        'worklet';
+        runOnJS(onGestureBegin)();
+      })
+      .onChange((event) => {
+        'worklet';
+        touchX.value = event.x;
+        translationX.value = startX.value + event.translationX;
+        gestureState.value = event.state;
+      })
+      .onEnd((event, success) => {
+        'worklet';
+        gestureState.value = event.state;
+
+        if (!success) {
+          runOnJS(onGestureAbort)();
+        }
+
+        const nextOpen =
+          (Math.abs(event.translationX) > SWIPE_MIN_OFFSET &&
+            Math.abs(event.translationX) > swipeMinVelocity) ||
+          Math.abs(event.translationX) > swipeMinDistance
+            ? drawerPosition === 'left'
+              ? // If swiped to right, open the drawer, otherwise close it
+                (event.velocityX === 0 ? event.translationX : event.velocityX) >
+                0
+              : // If swiped to left, open the drawer, otherwise close it
+                (event.velocityX === 0 ? event.translationX : event.velocityX) <
+                0
+            : open;
+
+        toggleDrawer(nextOpen, event.velocityX);
+        runOnJS(onGestureFinish)();
+      })
+      .activeOffsetX([-SWIPE_MIN_OFFSET, SWIPE_MIN_OFFSET])
+      .failOffsetY([-SWIPE_MIN_OFFSET, SWIPE_MIN_OFFSET])
+      .hitSlop(hitSlop)
+      .enabled(drawerType !== 'permanent' && swipeEnabled);
+
+    if (panGesture && configureGestureHandler) {
```

Comment:
It's important for this to be within the useMemo block since it could be mutating.